lib/port_scanner.py

```diff
@@ -0,0 +1,45 @@
+import queue
+import socket
+from threading import Thread, Lock
+from queue import Queue
+
+N_THREADS = 150
+
+queue = Queue()
+print_lock = Lock()
+
+def scan_port(host, port):
+    try:
+        sock = socket.socket()
+        sock.connect((host, port))
+    except:
+        pass
+    else:
+        with print_lock:
+            print(f"{host:15}:{port:5} is open")
```

Comment:
would be easier to just separate them by `\t` which would also auto align them, but that's also fine

Reply:
if it's file I'll leave it for now